test/e2e/e2e_test.go

```diff
@@ -1252,6 +1253,254 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
+		if ip := net.ParseIP(nodeIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", workerNodeInfo.name, nodeIP)
+		}
+		framework.Logf("the pod side node is %s and the source node ip is %s", workerNodeInfo.name, nodeIP)
+		podCIDR, err := getNodePodCIDR(workerNodeInfo.name)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", workerNodeInfo.name, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", workerNodeInfo.name, podCIDR)
+
+		// add loopback addresses used to validate all traffic is getting drained through the gateway
+		for lastOctet := 1; lastOctet <= ecmpRetry; lastOctet++ {
+			gwLoPrefix := fmt.Sprintf("%s%d/32", exGWRemoteIpPrefix, lastOctet)
+			// add the loopbacks to the first gateway container
+			_, err = runCommand("docker", "exec", gwContainer1, "ip", "address", "add", gwLoPrefix, "dev", "lo")
+			if err != nil {
+				framework.Failf("failed to add the loopback ip to dev lo on the test container: %v", err)
+			}
+			// add the same loopbacks to the second gateway container
+			_, err = runCommand("docker", "exec", gwContainer2, "ip", "address", "add", gwLoPrefix, "dev", "lo")
+			if err != nil {
+				framework.Failf("failed to add the loopback ip to dev lo on the test container %s: %v", gwContainer2, err)
+			}
+		}
+
+		// Create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPingPodName, workerNodeInfo.name, f.Namespace.Name, command)
+		// wait for pod setup to return a valid address
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			pingSrc, err = getPodAddress(srcPingPodName, f.Namespace.Name)
+			if err != nil {
+				framework.Logf("retrying ... error trying to get pod  %s address: %v", srcPingPodName, err)
+				return false, nil
+			}
+			validIP := net.ParseIP(pingSrc)
+			if validIP == nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		// fail the test if no address is ever retrieved
+		if err != nil {
+			framework.Failf("Error trying to get the pod IP address")
+		}
+		// add a host route on the gateways for return traffic to the pod
+		_, err = runCommand("docker", "exec", gwContainer1, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer1, err)
+		}
+		_, err = runCommand("docker", "exec", gwContainer2, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer2, err)
+		}
+		// install tcpdump to check for egress traffic to the gateway's loopback address
+		// since /proc/net/dev counters only record the ingress interface of the traffic
+		// a packet capture is needed to ensure traffic reaches the gateway loopback
+		_, err = runCommand("docker", "exec", gwContainer1, "yum", "-y", "install", "tcpdump")
```

Comment:
you could just use this docker image which already has tcpdump along with some other useful net tools:

docker.io/centos/tools:latest